src/cdk/constructs/domain-manager/resource/custom-domain.ts

```diff
@@ -0,0 +1,39 @@
+import { Topic } from '@aws-cdk/aws-sns'
+import { Construct } from '@aws-cdk/core'
+import {
+  CustomResource,
+  CustomResourceProvider
+} from '@aws-cdk/aws-cloudformation'
+import { MiraConfig } from '../../../..'
+import { DOMAIN_SUBSCRIPTION_TOPIC } from './route53-manager'
+
+export interface CustomDomainProps {
+  source: string
+  target: string
+}
+
+export class CustomDomain extends Construct {
+  domain: CustomResource
+
+  constructor (parent: Construct, props: CustomDomainProps) {
+    super(parent, 'CustomDomainResouce')
+
+    const topicName = MiraConfig.getBaseStackName(DOMAIN_SUBSCRIPTION_TOPIC)
```

Comment:
Could this become flaky or.. the semantics of using `getBaseStackName` throw me off a bit I think.  Other recent changes have used `calculateSharedResourceName`, so why use the `getBaseStackName`?  What happens if `DOMAIN_SUBSCRIPTION_TOPIC` changes or if we want more than one?  How does an app using Mira implement this?  This may be answered by other parts of code in this PR, still reading here :)

Reply:
I can't see any scenario when we want to have more than 1 `DOMAIN_SUBSCRIPTION_TOPIC`. Could you give an example?

If the value of that changes then that should be a breaking change in Mira I suppose and it requires a major version bump.

Reply:
Ok.. this one seems fine since it is just a topic in SNS.

Reply:
What happens though if we deploy across env's?  Does `getBaseStackName` output env?

Reply:
That's a good catch 👍 If we were to deploy multiple envs in to the same account this would cause problem. I've updated it in the latest commit.